src/Hook/SessionMigrationHook.php

```
@@ -0,0 +1,255 @@
+                'session_id' => SessionIdMasker::mask($sessionId),
+            ]);
+            return;
+        }
+
+        try {
+            $data = $this->pendingWrites[$sessionId];
+            $serializedData = $this->serializer->encode($data);
+            $targetId = $this->targetSessionId;
+
+            $this->logger->info('Starting session migration via hook', [
+                'old_session_id' => SessionIdMasker::mask($sessionId),
```

PHPMD (2.15.0)

Avoid using static access to class '\Uzulla\EnhancedRedisSessionHandler\Support\SessionIdMasker' in method 'afterWrite'. (undefined)

(StaticAccess)